packages/repository/src/repositories/kv.repository.ts

```diff
@@ -68,7 +64,7 @@ export interface KVRepository<T extends Model> extends Repository<T> {
    * @param options Options for the operation
    * @returns A promise of the TTL value
    */
-  ttl?(key: string, ttl: number, options?: Options): Promise<number>;
+  ttl?(key: string, options?: Options): Promise<number>;
 
   /**
    * Fetch all keys
```

Comment:
Loading all keys into memory is a foot-gun that opens doors to a DoS attack where the datasource is queried with a pattern that matches too many keys.

The only reason why I added the method `keys(): string[]`, was to work around the limitations of LB 3.x - the way how it was directly exposing juggler methods via REST APIs and a lack of support for streaming responses.

We don't have that limitation in LB4 because REST API is exposed via custom Controller classes that can work around missing support for streaming responses in LB4.

I am proposing to remove `keys(): string[]` API from LB4 and provide only one method based on AsyncIterator. Ideally, this single method should be called `keys` instead of `iterateKeys` for simplicity.

Until our REST layer supports AsyncIterators, we can provide a helper function to convert AsyncIterator into a static array that people can use in their controllers.

Thoughts?

I am ok to leave this change out of this pull request if you prefer, as long as we make it happen before the first public release of KeyValue feature in LB4.